libblkid/src/superblocks/xfs.c

```diff
@@ -199,7 +204,20 @@ static int xfs_verify_sb(struct xfs_super_block *ondisk)
 	    sbp->sb_dblocks < XFS_MIN_DBLOCKS(sbp))
 		return 0;
 
-	/* TODO: version 5 has also checksum CRC32, maybe we can check it too */
+	if ((sbp->sb_versionnum & 0x0f) == 5) {
+		if (!(sbp->sb_versionnum | XFS_SB_VERSION_MOREBITSBIT))
+			return 0;
+		if (!(sbp->sb_features2 | XFS_SB_VERSION2_CRCBIT))
+			return 0;
```

Comment:
Coverity scan reports this as an issue:
```
207         if ((sbp->sb_versionnum & 0x0f) == 5) {
>>>     CID 381139:  Integer handling issues  (CONSTANT_EXPRESSION_RESULT)
>>>     "sbp->sb_versionnum | 0x8000" is always 1/true regardless of the values of its operand. This occurs as the logical operand of "!".
208             if (!(sbp->sb_versionnum | XFS_SB_VERSION_MOREBITSBIT))
209                 return 0;
210             if (!(sbp->sb_features2 | XFS_SB_VERSION2_CRCBIT))
211                 return 0;
```

Is the '|' operator expected there? At first glance, I'd expect '&'.